packages/@atjson/renderer-commonmark/test/commonmark-test.ts

```diff
@@ -383,6 +383,23 @@ After all the lists
         expect(CommonMarkRenderer.render(document).trim()).toBe(md);
       });
 
+      // *a.*^b^ -> *a*.b if superscript annotations are unsupported
+      test('wrapping adjacent characters in an unknown annotation does not preserve boundary punctuation', () => {
+        let document = new OffsetSource({
+          content: '*a.*^b^',
+          annotations: [
+            { id: '1', type: '-atjson-parse-token', start: 0, end: 1, attributes: {} },
+            { id: '1', type: '-offset-italic', start: 0, end: 4, attributes: {} },
+            { id: '1', type: '-atjson-parse-token', start: 3, end: 4, attributes: {} },
+            { id: '1', type: '-atjson-parse-token', start: 4, end: 5, attributes: {} },
+            { id: '1', type: '-atjson-unknown', start: 4, end: 7, attributes: {} },
```

Comment:
O. This actually is a weird case. Subscript is _in_ the OffsetSource, but we can't render it out to markdown bc it isn't supported. We may need to cast the OffsetSource to a source that has only the support for markdown so we get `unknown` showing up correctly.

Reply:
This unit test assumes we have an unknown annotation, it doesn't make any assumptions about how we got the unknown annotation or what it represents.

Reply:
Gotcha!